src/overlays/actors/ovl_En_Yabusame_Mark/z_en_yabusame_mark.c

```diff
@@ -40,17 +29,221 @@ static ColliderQuadInit D_80B434F0 = {
+        arrowHitPos.z = this->collider.info.bumper.hitPos.z;
+
+        effectVelocity.y = 15.0f;
+
+        EffectSsHitMark_SpawnCustomScale(globalCtx, 0, 700, &arrowHitPos);
+
+        scoreIndex = 2;
+
+        distanceFromCenter.x = fabsf(D_80B435F0[this->subTypeIndex].x - arrowHitPos.x);
+        distanceFromCenter.y = fabsf(D_80B435F0[this->subTypeIndex].y - arrowHitPos.y);
+        distanceFromCenter.z = fabsf(D_80B435F0[this->subTypeIndex].z - arrowHitPos.z);
+
+        scoreDistance100 = D_80B4362C[this->typeIndex].unk_0;
+        scoreDistance60 = D_80B4362C[this->typeIndex].unk_4;
+        scoreDistance30 = D_80B4362C[this->typeIndex].unk_8;
+
+        if (distanceFromCenter.x > scoreDistance100 || distanceFromCenter.y > scoreDistance100 ||
+            distanceFromCenter.z > scoreDistance100) {
+            scoreIndex = 1;
+            if (distanceFromCenter.x > scoreDistance60 || distanceFromCenter.y > scoreDistance60 ||
+                distanceFromCenter.z > scoreDistance60) {
+                scoreIndex = 0;
+            }
+            if (distanceFromCenter.x > scoreDistance30 || distanceFromCenter.y > scoreDistance30 ||
+                distanceFromCenter.z > scoreDistance30) {
+                return;
+            }
+        }
+
+        osSyncPrintf("\n\n");
+        osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ posＸ ☆☆☆☆☆ %f\n" VT_RST, arrowHitPos.x);
+        osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ posＹ ☆☆☆☆☆ %f\n" VT_RST, arrowHitPos.y);
+        osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ posＺ ☆☆☆☆☆ %f\n" VT_RST, arrowHitPos.z);
+        osSyncPrintf(VT_FGCOL(YELLOW) "☆☆☆☆☆ hitＸ ☆☆☆☆☆ %f\n" VT_RST, D_80B435F0[this->subTypeIndex].x);
+        osSyncPrintf(VT_FGCOL(YELLOW) "☆☆☆☆☆ hitＹ ☆☆☆☆☆ %f\n" VT_RST, D_80B435F0[this->subTypeIndex].y);
+        osSyncPrintf(VT_FGCOL(YELLOW) "☆☆☆☆☆ hitＺ ☆☆☆☆☆ %f\n" VT_RST, D_80B435F0[this->subTypeIndex].z);
+        osSyncPrintf(VT_FGCOL(PURPLE) "☆☆☆☆☆ 小    ☆☆☆☆☆ %f\n" VT_RST, scoreDistance100);
+        osSyncPrintf(VT_FGCOL(PURPLE) "☆☆☆☆☆ 大    ☆☆☆☆☆ %f\n" VT_RST, scoreDistance60);
+        osSyncPrintf(VT_FGCOL(PURPLE) "☆☆☆☆☆ point ☆☆☆☆☆ %d\n" VT_RST, scoreIndex);
+        osSyncPrintf("\n\n");
+
+        if (scoreIndex == 2) {
+            func_800F5C64(0x922);
+        }
+        if (scoreIndex == 1) {
+            func_800F8D04(NA_SE_SY_TRE_BOX_APPEAR);
+            func_80078884(NA_SE_SY_TRE_BOX_APPEAR);
+        }
+        if (scoreIndex == 0) {
+            func_80078884(NA_SE_SY_DECIDE);
+        }
+        EffectSsExtra_Spawn(globalCtx, &arrowHitPos, &effectVelocity, &effectAccel, 5, scoreIndex);
+    }
+}
+#else
 #pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Yabusame_Mark/func_80B42F74.s")
+#endif
+
+void EnYabusameMark_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnYabusameMark* this = THIS;
+    Vec3f* vertexArray;
+    u32 arrayIndex;
+
+    this->actionFunc(this, globalCtx);
+    arrayIndex = this->typeIndex * 4;
```

Comment:
Is this temp definitely required to match? I feel like it should be possible to omit it.

Reply:
Could this be a 2d array, or does the codegen not allow that?

Reply:
this array access was really picky, I couldn't get the right combination of sll instructions with a 2d array, and the regalloc is wrong without the temp.

Reply:
im not sure of the context cause i didnt read the code yet, but smells like addition being done on the index instead of a 2d array

Reply:
i suspect that collisionVerticies (should be Vertices btw) might be an array of Quads. I'm not entirely sure why such a type hasn't been created yet.

Reply:
Oh yeah I missed that collisionVerticies wasn't renamed. It should be changed to `sCollisionVertices`